scene/gui/texture_button.cpp

```diff
@@ -294,41 +295,28 @@ void TextureButton::_bind_methods() {
 }
 
 void TextureButton::set_texture_normal(const Ref<Texture2D> &p_normal) {
-	if (normal == p_normal) {
-		return;
-	}
-
-	normal = p_normal;
-	queue_redraw();
-	update_minimum_size();
+	_set_texture(&normal, p_normal);
 }
 
 void TextureButton::set_texture_pressed(const Ref<Texture2D> &p_pressed) {
-	if (pressed == p_pressed) {
-		return;
-	}
-
-	pressed = p_pressed;
-	queue_redraw();
-	update_minimum_size();
+	_set_texture(&pressed, p_pressed);
 }
 
 void TextureButton::set_texture_hover(const Ref<Texture2D> &p_hover) {
-	if (hover == p_hover) {
-		return;
-	}
-
-	hover = p_hover;
-	queue_redraw();
-	update_minimum_size();
+	_set_texture(&hover, p_hover);
 }
 
 void TextureButton::set_texture_disabled(const Ref<Texture2D> &p_disabled) {
 	if (disabled == p_disabled) {
 		return;
 	}
-
+	if (disabled.is_valid()) {
+		disabled->disconnect(CoreStringNames::get_singleton()->changed, callable_mp((CanvasItem *)this, &CanvasItem::queue_redraw));
+	}
 	disabled = p_disabled;
+	if (disabled.is_valid()) {
+		disabled->connect(CoreStringNames::get_singleton()->changed, callable_mp((CanvasItem *)this, &CanvasItem::queue_redraw));
+	}
```

Comment:
Why is disabled texture handled differently?

Reply:
Because it does not affect the size, see above `get_minimum_size`

Reply:
That seems like an issue with the min size calculation. Why wouldn't it? Also, my god, that method is awful...

Reply:
Should this be handled separately? I'm not quite sure how this should be fixed or what the right approach is

Reply:
Should I pre-emptively make the disabled texture also update the size? Or should that be done in such a PR?

Reply:
I think you already do that with the current version of the PR. Any further changes can be done in a follow-up by anyone.